go/chat/signencrypt/codec.go

```diff
@@ -565,6 +566,59 @@ func OpenWhole(sealed []byte, encKey SecretboxKey, verifyKey VerifyKey, signatur
 	return append(output, moreOutput...), nil
 }
 
+// include verification of associated data. see
+// https://en.wikipedia.org/wiki/Authenticated_encryption#Authenticated_encryption_with_associated_data_(AEAD)
+
+type AEADMessage struct {
+	V int               `codec:"v" json:"v"`
+	A [sha512.Size]byte `codec:"a" json:"a"`
+	M []byte            `codec:"m" json:"m"`
+}
```

Comment:
Add a comment to A saying that it is the hash of some associated data? In general names could be more verbose (since you have codec directives for packing).

Reply:
was thinking that was obvious from `[sha512.Size]` but you're right. just a second...